frontend/taipy-gui/src/components/Taipy/Input.spec.tsx

```diff
@@ -128,6 +147,18 @@ describe("Input Component", () => {
             type: "SEND_UPDATE_ACTION",
         });
     });
+    it("should display visibility off icon when showPassword is true", async () => {
+        const { getByLabelText } = render(<Input value={"Test Input"} type="password" />);
+        const visibilityButton = getByLabelText("toggle password visibility");
+        fireEvent.click(visibilityButton);
+        const visibilityIcon = document.querySelector('svg[data-testid="VisibilityOffIcon"]');
+        expect(visibilityIcon).toBeInTheDocument();
+    });
+    it("should display visibility icon when showPassword is false", async () => {
+        const { getByLabelText } = render(<Input value={"Test Input"} type="password" />);
```

Comment:
not testing the icon's presence ?

Reply:
it should be in this case: `should display visibility off icon when showPassword is true`. That case lay right above this case actually

Reply:
updated